starlite/config/cors.py

```diff
@@ -12,33 +16,155 @@ class CORSConfig(BaseModel):
 
     allow_origins: List[str] = ["*"]
     """
-        List of origins that are allowed. Can use '*' in any component of the path, e.g. 'domain.*'.
-        Sets the 'Access-Control-Allow-Origin' header.
+    List of origins that are allowed. Can use '*' in any component of the path, e.g. 'domain.*'. Sets the 'Access-Control-Allow-Origin' header.
     """
-    allow_methods: List[str] = ["*"]
+    allow_methods: List[Union[Literal["*"], "Method"]] = ["*"]
     """
-        List of allowed HTTP methods.
-        Sets the 'Access-Control-Allow-Methods' header.
+    List of allowed HTTP methods. Sets the 'Access-Control-Allow-Methods' header.
     """
     allow_headers: List[str] = ["*"]
     """
-        List of allowed headers.
-        Sets the 'Access-Control-Allow-Headers' header.
+    List of allowed headers. Sets the 'Access-Control-Allow-Headers' header.
     """
     allow_credentials: bool = False
     """
-        Boolean dictating whether or not to set the 'Access-Control-Allow-Credentials' header.
+    Boolean dictating whether or not to set the 'Access-Control-Allow-Credentials' header.
     """
     allow_origin_regex: Optional[str] = None
     """
-        Regex to match origins against.
+    Regex to match origins against.
     """
     expose_headers: List[str] = []
     """
-        List of headers that are exposed via the 'Access-Control-Expose-Headers' header.
+    List of headers that are exposed via the 'Access-Control-Expose-Headers' header.
     """
     max_age: int = 600
     """
-        Response aching TTL in seconds, defaults to 600.
-        Sets the 'Access-Control-Max-Age' header.
+    Response aching TTL in seconds, defaults to 600. Sets the 'Access-Control-Max-Age' header.
     """
+
+    _compiled_regex: Union[EmptyType, Pattern] = PrivateAttr(Empty)
+    _is_allow_all_origins: Union[EmptyType, bool] = PrivateAttr(Empty)
+    _is_allow_all_methods: Union[EmptyType, bool] = PrivateAttr(Empty)
+    _is_allow_all_headers: Union[EmptyType, bool] = PrivateAttr(Empty)
+    _preflight_headers: Union[EmptyType, Dict[str, str]] = PrivateAttr(Empty)
+    _simple_headers: Union[EmptyType, Dict[str, str]] = PrivateAttr(Empty)
+
+    @validator("allow_headers", always=True)
+    def validate_allow_headers(cls, value: List[str]) -> List[str]:  # pylint: disable=no-self-argument
+        """Ensure that the trusted hosts have correct domain wildcards.
```

Comment:
Either I am too tired after a stressful week or this docstring doesn't make sense because it mentions trusted hosts.